### PR TITLE
Update metadata.yaml for all sources with sandbox connections

### LINKED ISSUE #42465: Live tests: automate test runs against sandbox accounts.
Now that we've added connections to Cloud for our sandbox accounts, we can move forward with automating live tests.

### Implementation
- Add a list of connection IDs against which to test to each connector's metadata.yaml. These sandboxes should come from the `integration-test-sandboxes` workspace [here](https://cloud.airbyte.com/workspaces/112412bb-78aa-4ca6-aace-ae0b83be2215/connections). We can match up the source:connections using `airbyte-api` and automatically populate the `metadata.yaml` for the connector.
- Update the GHA workflow for regression tests to add a pull request option. When regression tests are run in this context, they'll pass in the connection ID from a sandbox account.

### Acceptance criteria
- [ ] Live tests run automatically on PRs, against sandbox accounts only.
- [ ] The outcome of the test run is visible in a status check.

### PR DESCRIPTION
**Pull request by clnoll**
## What
Add test configuration info for live tests to metadata.yaml for all connections for which there is a sandbox connection in the [integration-test-sandboxes workspace](https://cloud.airbyte.com/workspaces/112412bb-78aa-4ca6-aace-ae0b83be2215/connections).

This will be used in https://github.com/airbytehq/airbyte/pull/42574 to automatically run live tests on PRs against sandbox accounts.

With https://github.com/airbytehq/airbyte/pull/42574 and https://github.com/airbytehq/airbyte/pull/42860, closes https://github.com/airbytehq/airbyte/issues/42465.

### CODE REVIEW

File: airbyte-integrations/connectors/source-bigcommerce/metadata.yaml [hunk 1]
```diff
@@ -35,6 +35,10 @@ data:
   # This connector is not enabled for OSS/Cloud
   #
   # connectorTestSuitesOptions:
+  #   - suite: liveTests
```
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-bigcommerce/metadata.yaml:38]: Intentionally keeping this commented for now?

**Review by natikgadzhi — APPROVED** [review]
I haven't verified connection IDs, but overall looks scripted :D 

Awesome work!

**Review by alafanechere — CHANGES_REQUESTED** [review]
You have to update the `metadata.yaml` model to declare a new enum value for `suite` and declare the `connectionOptions` property.
Changes have to be made [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/metadata_service/lib/metadata_service/models/src/ConnectorTestSuiteOptions.yaml#L10)

**Comment by clnoll** [comment]
Thanks @alafanechere I made those changes.

File: airbyte-integrations/connectors/source-bigcommerce/metadata.yaml (hunk 1 above)
Comment (clnoll) [reply airbyte-integrations/connectors/source-bigcommerce/metadata.yaml:38]: Yes, the sandbox connection isn't passing `check` and it has low usage.

**Review by alafanechere — COMMENTED** [review]
@clnoll I'll 👍 when we'll get passing QA checks on connectors, which will mean that metadata with the news fields are valid.

**Review by alafanechere — APPROVED** [review]
LGTM!

**Pull request merged**